optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/impl/domain/solution/JacksonUniqueIdGenerator.java

```diff
@@ -0,0 +1,54 @@
+package org.optaplanner.persistence.jackson.impl.domain.solution;
+
+import java.math.BigInteger;
+
+import com.fasterxml.jackson.annotation.ObjectIdGenerator;
+
+/**
+ * Similar in principle to {@link com.fasterxml.jackson.annotation.ObjectIdGenerators.UUIDGenerator},
+ * but without the overly long UUIDs.
+ */
+public final class JacksonUniqueIdGenerator extends com.fasterxml.jackson.annotation.ObjectIdGenerator<String> {
+
+    private final Class<?> scope;
+    private BigInteger nextValue = BigInteger.ZERO;
+
+    public JacksonUniqueIdGenerator() {
+        this.scope = Object.class;
+    }
+
+    @Override
+    public Class<?> getScope() {
+        return scope;
+    }
+
+    @Override
+    public boolean canUseFor(ObjectIdGenerator<?> gen) {
+        return (gen.getClass() == getClass());
+    }
+
+    @Override
+    public ObjectIdGenerator<String> forScope(Class<?> scope) {
+        return this;
+    }
+
+    @Override
+    public ObjectIdGenerator<String> newForSerialization(Object context) {
+        return this;
+    }
+
+    @Override
+    public IdKey key(Object key) {
+        if (key == null) {
+            return null;
+        }
+        return new IdKey(getClass(), null, key);
+    }
+
+    @Override
+    public synchronized String generateId(Object forPojo) {
+        BigInteger result = nextValue;
+        nextValue = nextValue.add(BigInteger.ONE);
+        return result.toString(16); // Shorten possibly large numbers to hexadecimal representation.
+    }
```

Comment:
Can we possibly make the critical section shorter?

```suggestion
    public String generateId(Object forPojo) {
        synchronized {
            nextValue = nextValue.add(BigInteger.ONE);
        }
        return nextValue.toString(16); // Shorten possibly large numbers to hexadecimal representation.
    }
```
Of course, the `nextValue` field has to be initialized to `BigInteger.ONE`.